From 8651bfa84a73b7a5284e4abe062b49d66fab68c7 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Fri, 5 Feb 2016 02:39:21 -0500
Subject: [PATCH] policy/modules/services/watchdog: allow watchdog_t to create
 /var/log/watchdog

* Patches ported from Fedora 22:
  Reference sources: selinux-policy-3.13.1-128.21.fc22.src.rpm
  Changes come from: policy-f22-services.patch

Fixes:
  avc: denied { create } for pid=372 comm="watchdog" name="watchdog" \
  scontext=system_u:system_r:watchdog_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:var_log_t:s0 \
  tclass=dir

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/watchdog.te | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/policy/modules/services/watchdog.te b/policy/modules/services/watchdog.te
index 5b3c88893..1cf8b4c84 100644
--- a/policy/modules/services/watchdog.te
+++ b/policy/modules/services/watchdog.te
@@ -30,8 +30,9 @@ allow watchdog_t self:fifo_file rw_fifo_file_perms;
 allow watchdog_t self:rawip_socket create_socket_perms;
 allow watchdog_t self:tcp_socket { accept listen };
 
-allow watchdog_t watchdog_log_t:file { append_file_perms create_file_perms setattr_file_perms };
-logging_log_filetrans(watchdog_t, watchdog_log_t, file)
+manage_files_pattern(watchdog_t, watchdog_log_t, watchdog_log_t)
+manage_dirs_pattern(watchdog_t, watchdog_log_t, watchdog_log_t)
+logging_log_filetrans(watchdog_t, watchdog_log_t, { dir file })
 
 manage_files_pattern(watchdog_t, watchdog_runtime_t, watchdog_runtime_t)
 files_runtime_filetrans(watchdog_t, watchdog_runtime_t, file)
@@ -71,6 +72,7 @@ files_read_all_runtime_files(watchdog_t)
 fs_unmount_xattr_fs(watchdog_t)
 fs_getattr_all_fs(watchdog_t)
 fs_search_auto_mountpoints(watchdog_t)
+fs_search_tmpfs(watchdog_t)
 
 auth_append_login_records(watchdog_t)
 
-- 
2.17.1

